### DISCUSSION TITLE
Lines appear between sprites after migrating to 0.6.

### DISCUSSION THREAD

**Discussion by automeowtion** [bevyengine/bevy#3748] [Q&A]
Lines appear if `cam.orthographic_projection.scale` is not 1.0.
Sprites are spawned through `SpriteSheetBundle`.
The lines may be horizontal or vertical depending on the sprites but always appear at sprite boundaries.
Camera is `OrthographicCameraBundle::new_2d()` with only scale adjusted.
In the picture below, sprite size is 32x32 pixels and camera scale is 0.7.
<img width="231" alt="Screen Shot 2022-01-22 at 10 07 12 PM" src="https://user-images.githubusercontent.com/69330666/150666026-c61ea580-a3d3-44f3-98f8-98bf688c6496.png">

**Comment by StarArawn** [comment]
@automeowtion Hi! You found an issue that's quite common with atlas rendering. There are a few ways of solving this:
1. Stop using atlases.
2. Add spacing between each sprite in the atlas.
3. Try using nearest neighbor filtering instead of linear filtering for the textures.

I would recommend option 1. A better solution would be to use a texture array. Each texture in the array can be sampled and filtered independently of the other textures, however bevy doesn't have support for texture arrays out of the box. You'll need to setup a custom render pass and create the texture array manually there is a bit of work and know how that goes into that. A simpler approach is to go with option 2 or 3. Option 2 is your best bet between those choices.

**Reply by colepoirier** [reply]
@superdump Are you working on adding support for texture arrays?

**Reply by superdump** [reply]
I haven't really been doing anything with 2D. For the PBR stuff, I have a branch where I'm trying to make the core shader functionality callable and reusable. So then you can use completely custom bindings to get all the data that the fragment shader needs, like the base colour, normal maps, emissive, metallic, roughness, etc properties, and then call the main PBR shader that uses those. As a proof of concept, I was going to port the old array texture example to use it as the array textures (as in `texture_2d_array<f32>` in WGSL) would be a different binding than a plain 2D base colour texture.

**Reply by StarArawn** [reply]
It's probably worth mentioning that currently texture arrays don't work on webGL2. They should work but there appears to be a bug in wgpu that's affecting that.

**Reply by automeowtion** [reply]
Ty for the very helpful reply!
> 3. Try using nearest neighbor filtering instead of linear filtering for the textures.

I am trying to learn as much as possible about each of the three options. With option 3, I am guessing that the following line needs to be added somewhere:
`image.sampler_descriptor.min_filter = FilterMode::Nearest;`

At app start, TextureAtlas are created from png files. Then, I print the SamplerDescriptor of the texture atlases. And it appears that the filters are already set to `Nearest` by default:
```
[src/main.rs:215] &image.sampler_descriptor = SamplerDescriptor {
    label: None,
    address_mode_u: ClampToEdge,
    address_mode_v: ClampToEdge,
    address_mode_w: ClampToEdge,
    mag_filter: Nearest,
    min_filter: Nearest,
    mipmap_filter: Nearest,
    lod_min_clamp: 0.0,
    lod_max_clamp: 3.4028235e38,
    compare: None,
    anisotropy_clamp: None,
    border_color: None,
}
```

**Comment by yopox** [comment]
In Bevy 0.10, insert the `Msaa::Off` resource to fix the issue!
